shopfloor/actions/data.py

```diff
@@ -11,13 +11,37 @@ class DataAction(Component):
 
     @ensure_model("stock.location")
     def location(self, record, **kw):
-        return self._jsonify(
-            record.with_context(location=record.id), self._location_parser, **kw
-        )
+        parser = self._location_parser
+        data = self._jsonify(record.with_context(location=record.id), parser, **kw)
+        if "with_operations_progress_info" in kw:
+            operations_done, operations_to_do = self._get_location_operations_progress(
+                record
+            )
+            data.update(
+                {
+                    "operations_done": operations_done,
+                    "operations_to_do": operations_to_do,
+                }
+            )
+        return data
 
     def locations(self, record, **kw):
         return self.location(record, multi=True)
 
+    def _get_location_operations_progress(self, location):
+        location_pickings = self.env["stock.picking"].search(
+            [("location_id", "=", location.id), ("state", "=", "assigned")]
```

Comment:
You need to collect the lines from the processed location. The location of the picking can be a parent of the processed location and cannot be used in the search.

stock.move.line search:

location_id = location.id
state in partially_available, assigned
picking_id.state = assigned